### ISSUE TITLE
Feature Request: Action to Bookmark All Tabs

### ISSUE THREAD

**Issue by sahil48** [mozilla-mobile/firefox-ios#6110]
It would be very helpful to bookmark all open tabs to a folder of my choice.

**Comment by elemanhillary** [comment]
Can i work on this

**Comment by elemanhillary** [comment]
As in the feature is already implemented and ready for a push

**Comment by data-sync-user** [comment]
➤ Ashley Thomas commented:

Thank you for following up. This was delayed because we've been looking at a major change to the tabs tray #7308 .  [~bpitoyo@mozilla.com] Can you take a look at the UX for this and see what you think?  I'm wondering how it might work with the new tabs design. Here's the PR https://github.com/mozilla-mobile/firefox-ios/pull/6126

**Comment by lolrepeatlol** [comment]
IIRC, I'm pretty sure Firefox for Android has a feature called Collections which is essentially what this is. Ideally it should also be called Collections on iOS and sync to Android.

**Comment by sahil48** [comment]
I am unfamiliar with collections, I wanted bookmarking of all tabs so that they also sync's with my Firefox Desktop Bookmarks. I oftentime browse on my phone, bookmark all tabs that are still of interest occasionally, and then read through them on a bigger screen. The ability to put bookmarks in a nested folder structure always appealed to me as well.

**Comment by elemanhillary** [comment]
if there's any change i can work on it since i authored the PR

**Comment by brampitoyo** [comment]
@athomasmoz Before we design or implement anything, let’s wait until the new tabs tray (#7308) landed.

That change will of course include the ability to auto-sort tabs by date (chronologically), but it will also change the way tab tray work.

Firstly, it will allow easy switching between Normal/Private/Synced Tabs (today, Synced Tabs are located inside the Library).

Secondly, it will allow the selection of multiple tabs. After select multiple is implemented, mass-tab operations can quite naturally sit on top of it: share multiple tabs, bookmark multiple tabs, etc.

**Comment by sahil48** [comment]
Has the new tab tray now landed to allow introducing this feature?

**Comment by fourquarts** [comment]
Came here to add my vote for this feature. No other browser on iOS has this, so far as I am aware — not Safari, not Brave, not Firefox. The "tabs in the cloud" feature is nice, but it's only really good for accessing one tab at a time. I would like to be able to save all iOS tabs as bookmarks and then sort through what I want to keep on my desktop computer. Please, please add this soon. It really is a needed feature.

**Comment by sahil48** [comment]
@fourquarts Safari for iOS actually does have this feature (https://www.groovypost.com/howto/bookmark-all-open-safari-tabs-at-once-in-ios/). It makes it all the more significant that Firefox for iOS has not implemented this feature. I prefer Firefox Mobile because I can work with bookmarked tabs on the Desktop better, and I like the browser better in general, but really do need this feature to use it as my main browser on my phone and tablet.

**Comment by elemanhillary** [comment]
@sahil48 they left below comment on the PR that solves this issue
Closing this for now as it needs some UX input before we can merge.

**Comment by fourquarts** [comment]
Wow - I did not know that about Safari. Thank you!

On Sep 11, 2021, at 7:51 AM, sahil48 ***@***.***> wrote:

﻿
@fourquarts Safari for iOS actually does have this feature (https://www.groovypost.com/howto/bookmark-all-open-safari-tabs-at-once-in-ios/). It makes it all the more significant that Firefox for iOS has not implemented this feature. I prefer Firefox Mobile because I can work with bookmarked tabs on the Desktop better, and I like the browser better in general, but really do need this feature to use it as my main browser on my phone and tablet.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by Codplay** [comment]
@brampitoyo @dnarcese Can you provide further information about why this is not being implemented? The new tabs tray has most definitely landed, but unlike as was mentioned it doesn't support any multi-select as far as I can tell (only drag and drop to rearrange).

It seems like an stupidly simple feature to implement, especially when the actual work has been done by @elemanhillary for over A YEAR AND A HALF now.

**Comment by Codplay** [comment]
@brampitoyo @dnarcese - can we get an update on this? I see it has a JIRA task linked but obviously can’t see the status of that. You say it needs UX input - has that been assigned to anyone? Considering that the issue has been functionally fixed by another user, it’s absurd this issue is still a concern after TWO YEARS!!!

**Comment by lolrepeatlol** [comment]
> @brampitoyo @dnarcese - can we get an update on this? I see it has a JIRA task linked but obviously can’t see the status of that. You say it needs UX input - has that been assigned to anyone? Considering that the issue has been functionally fixed by another user, it’s absurd this issue is still a concern after TWO YEARS!!!

just because a feature request/issue is filed, does not mean it needs to get fixed or alleviated immediately. this is true of all projects. yelling at developers will not help. please stop

**Comment by lmarceau** [comment]
Thanks @lolrepeatlol 
Indeed as much as we want, our team cannot enable all requests by contributors. Rest assured thought that all **feature requests** are duly considered by the product team. 

On another note, @ brampitoyo isn't at Mozilla anymore so tagging him here has no effect 👍 Thank you for your understanding.

**Comment by sahil48** [comment]
Thank you @lmarceau . Is there someone else who can be assigned the issues that were assigned to brampitoyo so that they are reviewed and addressed.  If an employee leaves, and their issues are not reassigned, there is a false sense that the issues are under someone's review and responsibility who can be contacted when in actuality they are not. This is a process improvement that I think could be very well-worth it for the product team.

**Comment by dnarcese** [comment]
@sahil48 Yes, we are working on getting a new designer to look at this and will prioritize accordingly

**Comment by data-sync-user** [comment]
➤ Michael Verdi commented:

We looked at a few options and requests around making bookmarks easier and more discoverable. We added recent bookmarks to the homepage, we made a single menu so there’s only one place to look for bookmarks and we moved mobile bookmarks to the top level of the bookmarks section. But with the inactive tabs feature and a possible tab grouping feature we didn’t think this would work well or be helpful for a large group of people so we decided not to do this.

**Comment by asclepiadae** [comment]
I am struggling to see how the inactive tabs and other usability features render this request irrelevant. It doesn't matter how few taps it takes - to bookmark (or send to a desktop OS) a single tab is, I still have to long press on the tab thumbnail, and then it's at minimum two taps to save or send each tab, plus closing.

So for 100 tabs that's **400 steps**. A bookmark all tabs (or send all tabs to device) feature, along with a close all tabs, would take that down to perhaps five steps depending on how it was implemented.

This is a feature that competing mobile browsers **_have_** determined to be relevant, and included as a feature.

This is a feature that brings parity of operation from the desktop experience to mobile - that alone, given no OS limitation (lack of API for example), should at least determine that it should be somewhere on an outstanding issues list (as a low priority, sure!)

This is also a feature that has had code provided by a member of the community (although how that code still functions is unknown).

At the end of the day this is not a request that will require weeks of development from a team. (At this point the only reason I am not writing the code myself is unfamiliarity with Swift and no Mac for a development environment).

It could even be included without **_any_** UI changes, in the settings menu! ![image](https://github.com/mozilla-mobile/firefox-ios/assets/2826435/677cfefc-03d5-49cd-a3ab-87e1e8107238)

If there are _technical_ limitations on this request, could those be expanded upon?

Thanks!
